src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt

```diff
@@ -35,7 +38,12 @@ class OrderService(val db: OrderRepository, val synqService: SynqOrderService) {
             return ResponseEntity.badRequest().build()
         }
 
-        synqService.createOrder(payload.toOrder().toSynqPayload())
+        // Handle any edge-cases
+        val synqResponse = synqService.createOrder(payload.toOrder().toSynqPayload())
+        if (!synqResponse.statusCode.isSameCodeAs(HttpStatus.CREATED)) {
+            throw ServerErrorException("Unexpected error with SynQ", null)
```

Comment:
Need to talk about it with Tom, how should we handle it

Reply:
To clarify this edge-case only happens if the `createOrder` call doesn't throw, but still returns something that isn't a new order. Will update the comment accordingly

Reply:
Yeah, just thinking about what to give back to caller and how to handle it in DB and storage systems, eg cleanup